x-pack/plugins/rule_registry/server/utils/create_lifecycle_rule_type_factory.ts

```diff
@@ -168,57 +168,59 @@ export const createLifecycleRuleTypeFactory: CreateLifecycleRuleTypeFactory = ({
         });
       }
 
-      const eventsToIndex = allAlertIds.map((alertId) => {
-        const alertData = alertsDataMap[alertId];
+      const eventsToIndex = ruleDataClient.isWriteEnabled()
```

Comment:
Should we short-circuit here or just around bulk? If we do the latter I suppose we have to set `trackedAlerts` in the rule state to an empty array. But that might have the benefit of being more explicit.

Reply:
hm, might be a valid point. I'll try to see if reformulating leads to better readability.